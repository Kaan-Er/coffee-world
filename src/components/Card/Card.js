import React from "react";

const Card = ({ coffee }) => {
  const { title, description, ingredients } = coffee;
  return (
    <div className="card">
      <h2 className="card-title" data-testid="header">
        {title}
      </h2>
      <p className="card-description">{description}</p>
      <p className="card-ingredients">Ingredients: {ingredients.join(", ")}</p>
    </div>
  );
};

export default Card;
