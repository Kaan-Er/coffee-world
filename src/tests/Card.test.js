import React from "react";
import renderer from "react-test-renderer";
import Card from "../components/Card/Card";

it("Card renders correctly", () => {
  const coffee = {
    title: "Black",
    description:
      "Black coffee is as simple as it gets with ground coffee beans steeped in hot water, served warm. And if you want to sound fancy, you can call black coffee by its proper name: cafe noir.",
    ingredients: ["Coffee"],
    category: "hot",
    id: 1,
  };
  const tree = renderer.create(<Card coffee={coffee} />).toJSON();
  expect(tree).toMatchSnapshot();
});
