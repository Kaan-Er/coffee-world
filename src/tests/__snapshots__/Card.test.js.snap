// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="card"
>
  <h2
    className="card-title"
    data-testid="header"
  >
    Black
  </h2>
  <p
    className="card-description"
  >
    Black coffee is as simple as it gets with ground coffee beans steeped in hot water, served warm. And if you want to sound fancy, you can call black coffee by its proper name: cafe noir.
  </p>
  <p
    className="card-ingredients"
  >
    Ingredients: 
    Coffee
  </p>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <h2
    className="card-title"
    data-testid="header"
  >
    Black
  </h2>
  <p
    className="card-description"
  >
    Black coffee is as simple as it gets with ground coffee beans steeped in hot water, served warm. And if you want to sound fancy, you can call black coffee by its proper name: cafe noir.
  </p>
  <p
    className="card-ingredients"
  >
    Ingredients: 
    Coffee
  </p>
</div>
`;
